components/powersensor/powersensor.go

```diff
@@ -0,0 +1,91 @@
+// Package powersensor defines the interfaces of a powersensor
+package powersensor
+
+import (
+	"context"
+	"strings"
+
+	pb "go.viam.com/api/component/powersensor/v1"
+
+	"go.viam.com/rdk/components/sensor"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/robot"
+)
+
+func init() {
+	resource.RegisterAPI(API, resource.APIRegistration[PowerSensor]{
+		RPCServiceServerConstructor: NewRPCServiceServer,
+		RPCServiceHandler:           pb.RegisterPowerSensorServiceHandlerFromEndpoint,
+		RPCServiceDesc:              &pb.PowerSensorService_ServiceDesc,
+		RPCClient:                   NewClientFromConn,
+	})
+}
+
+// SubtypeName is a constant that identifies the component resource API string "power_sensor".
+const SubtypeName = "power_sensor"
+
+// API is a variable that identifies the component resource API.
+var API = resource.APINamespaceRDK.WithComponentType(SubtypeName)
+
+// Named is a helper for getting the named PowerSensor's typed resource name.
+func Named(name string) resource.Name {
+	return resource.NewName(API, name)
+}
+
+// A PowerSensor reports information about voltage, current and power.
+type PowerSensor interface {
+	sensor.Sensor
+	Voltage(ctx context.Context, extra map[string]interface{}) (float64, bool, error)
+	Current(ctx context.Context, extra map[string]interface{}) (float64, bool, error)
+	Power(ctx context.Context, extra map[string]interface{}) (float64, error)
+}
+
+// FromDependencies is a helper for getting the named PowerSensor from a collection of
+// dependencies.
+func FromDependencies(deps resource.Dependencies, name string) (PowerSensor, error) {
+	return resource.FromDependencies[PowerSensor](deps, Named(name))
+}
+
+// FromRobot is a helper for getting the named PowerSensor from the given Robot.
+func FromRobot(r robot.Robot, name string) (PowerSensor, error) {
+	return robot.ResourceFromRobot[PowerSensor](r, Named(name))
+}
+
+// NamesFromRobot is a helper for getting all PowerSensor names from the given Robot.
+func NamesFromRobot(r robot.Robot) []string {
+	return robot.NamesByAPI(r, API)
+}
+
+// Readings is a helper for getting all readings from a PowerSensor.
+func Readings(ctx context.Context, g PowerSensor, extra map[string]interface{}) (map[string]interface{}, error) {
+	readings := map[string]interface{}{}
+
+	vol, _, err := g.Voltage(ctx, extra)
```

Comment:
I wasn't really sure if I should be doing something with the isAC boolean in the readings function if any reviewers have opinions?

Reply:
I think we should return at least one of them, since it's data that has been deemed important enough to return for any specific reading.

If the `isAc` result from calling `g.Voltage` differs from the one from calling `g.Current`, very weird things are happening. Maybe we just return the one from `g.Voltage` and don't bother checking if it matches the others? Maybe we return all of them just in case. Maybe we check that they all match and return an error if they don't? This points to a bit of a smell in the interface, because it's possible to represent impossible state. but I suspect at this point it's too late to change that (I don't even know what we'd change it to), and we should just keep going with the current version.

Reply:
I'm leaning towards just returning one of them, @randhid do you have an opinion of which of these options is best?

Reply:
just return one, someone returning a mismatch is purely a driver error I think. This is also easy to change later.